anomalous-value/src/agent.js

```diff
@@ -40,45 +28,97 @@ function createAlert(log) {
-      // if we have seen this before, check for anomalous value
-      const average = rollingEventData[log.args.reserve].getAverage();
-      const standardDeviation = rollingEventData[log.args.reserve].getStandardDeviation();
+function provideInitialize(data) {
+  return async function initialize() {
+    /* eslint-disable no-param-reassign */
+    const { LendingPool: lendingPoolConfig } = config.contract;
+    const abi = getAbi(lendingPoolConfig.abiFile);
+
+    // create ethers interface object
+    const iface = new ethers.utils.Interface(abi);
+
+    // events we are interested in
+    const eventFragments = [
+      iface.getEvent('Borrow'),
+      iface.getEvent('Deposit'),
+      iface.getEvent('Repay'),
+      iface.getEvent('Withdraw'),
+    ];
+
+    // assign configurable fields
+    data.address = lendingPoolConfig.address;
+    data.protocolName = config.protocolName;
+    data.protocolAbbrev = config.protocolAbbreviation;
+    data.developerAbbrev = config.developerAbbreviation;
+    data.eventFragments = eventFragments;
+    data.type = lendingPoolConfig.type;
+    data.severity = lendingPoolConfig.severity;
+
+    // create rolling math object structure
+    data.rollingEventData = {};
+    data.windowSize = lendingPoolConfig.windowSize;
+    data.standardDeviations = lendingPoolConfig.standardDeviations;
+    /* eslint-enable no-param-reassign */
+  };
+}
 
-      // limit is set from agent-config.json file
-      const limit = standardDeviation.times(anomalousValue.standardDeviations);
-      const delta = amount.minus(average).absoluteValue();
+function provideHandleTransaction(data) {
+  return async function handleTransaction(txEvent) {
+    const {
+      developerAbbrev,
+      protocolName,
+      protocolAbbrev,
+      address,
+      eventFragments,
+      type,
+      severity,
+      rollingEventData,
+      windowSize,
+      standardDeviations,
+    } = data;
+    const findings = [];
+    const parsedLogs = txEvent.filterLog(eventFragments, address);
+
+    // loop over each eventLog, checking for anomalous value
+    parsedLogs.forEach((log) => {
+      // rolling math requires bignumber.js style BigNumbers
+      const amount = new BigNumber(log.args.amount.toHexString());
+
+      // if we haven't seen this reserve yet, initialize it
+      if (!rollingEventData[log.args.reserve]) {
+        rollingEventData[log.args.reserve] = new RollingMath(windowSize);
+      }
 
-      // if instance is outside the standard deviation, report
-      if (delta.isGreaterThan(limit)) {
-        findings.push(createAlert(log));
+      // only process data for alerts if we have seen a significant number of blocks
+      if (rollingEventData[log.args.reserve].getNumElements() >= windowSize) {
+        // if we have seen this before, check for anomalous value
+        const average = rollingEventData[log.args.reserve].getAverage();
+        const standardDeviation = rollingEventData[log.args.reserve].getStandardDeviation();
```

Comment:
Nit: It may be a bit more readable to destructure `reserve`
```suggestion
      const { reserve } = logs.args;
      if (!rollingEventData[reserve]) {
        rollingEventData[reserve] = new RollingMath(windowSize);
      }

      // only process data for alerts if we have seen a significant number of blocks
      if (rollingEventData[reserve].getNumElements() >= windowSize) {
        // if we have seen this before, check for anomalous value
        const average = rollingEventData[reserve].getAverage();
        const standardDeviation = rollingEventData[reserve].getStandardDeviation();
```